wcfsetup/install/files/lib/system/endpoint/controller/core/gridViews/GetGridView.class.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace wcf\system\endpoint\controller\core\gridViews;
+
+use Laminas\Diactoros\Response\JsonResponse;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use wcf\http\Helper;
+use wcf\system\endpoint\GetRequest;
+use wcf\system\endpoint\IController;
+use wcf\system\exception\PermissionDeniedException;
+use wcf\system\exception\UserInputException;
+use wcf\system\gridView\AbstractGridView;
+
+/**
+ * API endpoint for the rendering of a grid view in a dialog.
+ *
+ * @author Olaf Braun
+ * @copyright 2001-2025 WoltLab GmbH
+ * @license GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @since 6.2
+ */
+#[GetRequest('/core/grid-views/render')]
+final class GetGridView implements IController
+{
+    #[\Override]
+    public function __invoke(ServerRequestInterface $request, array $variables): ResponseInterface
+    {
+        $parameters = Helper::mapApiParameters($request, GetGridViewParameters::class);
+
+        if (!\is_subclass_of($parameters->gridView, AbstractGridView::class)) {
+            throw new UserInputException('gridView', 'invalid');
+        }
+
+        $view = new $parameters->gridView(...$parameters->gridViewParameters);
+        // @phpstan-ignore function.alreadyNarrowedType, instanceof.alwaysTrue
+        \assert($view instanceof AbstractGridView);
+
+        if (!$view->isAccessible()) {
+            throw new PermissionDeniedException();
+        }
+
+        $view->setPageNo($parameters->pageNo);
+        if ($parameters->sortField) {
+            $view->setSortField($parameters->sortField);
+        }
+        if ($parameters->sortOrder) {
+            $view->setSortOrder($parameters->sortOrder);
+        }
+
+        if ($parameters->filters !== []) {
+            $view->setActiveFilters($parameters->filters);
+        }
+
+        return new JsonResponse([
+            'gridView' => $view->render(),
+        ]);
+    }
+}
+
+/** @internal */
+final class GetGridViewParameters
+{
+    public function __construct(
+        /** @var non-empty-string */
+        public readonly string $gridView,
+        public readonly int $pageNo,
+        public readonly string $sortField,
+        public readonly string $sortOrder,
+        /** @var string[] */
+        public readonly array $filters,
+        /** @var string[] */
```

Comment:
```suggestion
        /** @var array<string, mixed> */
```

String keys result in the parameters being mapped using named parameters in line 35.